src/Components/UI/Navigation/MobileNavigation/MobileNavigationBar.jsx

```diff
@@ -0,0 +1,100 @@
+import NavbarLoginButton from "../NavbarLoginButton/NavbarLoginButton";
+import SearchBar from "../SearchBar/SearchBar";
+import logoIcon from "../../../../Data/Logo/icon.svg";
+import NavbarAvatar from "../Avatar/NavbarAvatar";
+import { useAuth, useFilter, useModal } from "../../../../Context";
+import { Link, useLocation, useNavigate } from "react-router-dom";
+import "./MobileNavigationBar.css";
+
+export const MobileNavigationBar = () => {
+  const { auth, authDispatch, showProfileMenu, setShowProfileMenu } = useAuth();
+  const { filterDispatch, searchInput, setSearchInput } = useFilter();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const { setAlertText, setShowAlert, setShowNavMenu } = useModal();
+
+  const onSearchSubmitHandler = (e) => {
+    e.preventDefault();
+    filterDispatch({ type: "bySearch", payload: searchInput });
+    navigate(`/videos/search?query=${searchInput}`);
+    setSearchInput("");
+  };
+
+  const onSearchInputHandler = (e) => {
+    setSearchInput(e.target.value);
+  };
+
+  const logoutClickHandler = () => {
+    authDispatch({ type: "logout" });
+    setAlertText("Logged out successfully");
+    setShowAlert(true);
+    if (location.pathname.includes("playlist" || "account" || "settings")) {
+      navigate("/videos");
+    }
+  };
+
+  const toggleProfileMenu = () => {
+    setShowProfileMenu((show) => !show);
+  };
+
+  const toggleAccountNavMenu = () => {
+    setShowNavMenu((show) => !show);
+  };
+
+  return (
+    <nav className="mobile-navigation-bar">
+      <div className="nav-header">
+        <div onClick={toggleAccountNavMenu} className="hamburger">
+          <i className="fas fa-bars"></i>
+        </div>
+        <Link to="/">
+          <img className="logo" src={logoIcon} alt="logo" />
+        </Link>
+      </div>
+      <SearchBar
+        searchWrapper="search-container"
+        micIcon="hide"
+        searchIcon="fas fa-search"
+        placeholder="Search"
+        onChange={onSearchInputHandler}
+        onSubmit={onSearchSubmitHandler}
+        value={searchInput}
+      />
+      <div className="nav-bar-btns">
+        {!auth.login && (
+          <NavbarLoginButton
+            label={auth.login ? "Logout" : "Login"}
+            btnClassName="btn primary-btn-md"
+          />
+        )}
+      </div>
+      <div className="profile-theme-toggle">
+        {auth.login && (
```

Comment:
instead of using short circuiting you could have used ternary operator

Reply:
I am following the practice to use logical && when the condition only returns a truthy or falsy value.